src/core/thread/mle.cpp

```diff
@@ -2792,7 +2792,13 @@ void Mle::HandleUdpReceive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
 
     aesCcm.Finalize(tag);
 #ifndef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
-    VerifyOrExit(memcmp(messageTag, tag, sizeof(tag)) == 0, error = kErrorSecurity);
+    // Intentionally ignore security check failures for broadcast MLE messages because it can be common to receive
+    // such kind of messages from adjacent Thread networks.
+    VerifyOrExit(memcmp(messageTag, tag, sizeof(tag)) == 0,
+                 error = ((aMessageInfo.GetSockAddr().IsMulticast() &&
+                           aMessageInfo.GetThreadLinkInfo()->GetPanId() == Mac::kPanIdBroadcast)
+                              ? kErrorNone
+                              : kErrorSecurity));
```

Comment:
This works fine as is. 
A suggestion here which may help us in future in other cases where we may want to skip logging error.
- How about adding a `bool skipLoggingError = false`
   - Then we check here if the MLE msg is broadcast and and security fails, we set `skipLoggingError` to `true`
- At the end where we want to log error we check this flag.